src/test/java/software/amazon/nio/spi/s3/util/StrictPosixGlobPathMatcherTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.nio.spi.s3.util;
+
+import org.junit.jupiter.api.Test;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class StrictPosixGlobPathMatcherTest {
+
+    @Test
+    void testSimpleGlobMatching() {
+        StrictPosixGlobPathMatcher matcher = new StrictPosixGlobPathMatcher("*.txt");
+        
+        assertThat(matcher.matches(Paths.get("file.txt"))).isTrue();
+        assertThat(matcher.matches(Paths.get("file.csv"))).isFalse();
+        assertThat(matcher.matches(Paths.get("dir/file.txt"))).isFalse(); // Should not match across directories
```

Comment:
Could be helpful when it fails and also gives context as the comment.
```suggestion
        assertThat(matcher.matches(Paths.get("dir/file.txt"))).as("Should not match across directories").isFalse();
```